src/Appwrite/Platform/Modules/Proxy/Http/Rules/Redirect/Create.php

```diff
@@ -114,6 +114,13 @@ public function action(string $domain, string $url, int $statusCode, string $res
             throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'This domain name is not allowed. Please use a different domain.');
         }
 
+        if (\str_ends_with(\strtolower($domain), '.appwrite.network')) {
+            $subdomain = \str_replace('.appwrite.network', '', strtolower($domain));
+            if (\str_contains($subdomain, '.')) {
+                throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, 'Sub-subdomains are not allowed for appwrite.network. Only one level of subdomain is permitted.');
+            }
+        }
```

Comment:
I see we have same code in many places.
It might be better to create custom Validator for this (check other Appwrite custom validators, I am sure there are a few), and here just use the validator.

this keeps critical logic in single place, easier to adjust in future, if needed.